lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb

```diff
@@ -123,57 +133,56 @@ def upload_binary(app_id, binary_path, platform)
       # Takes at least POLLING_INTERVAL_SECONDS between polling get_upload_status
       #
       # args
-      #   project_number - Firebase project number
       #   app_id - Firebase app ID
       #   binary_path - Absolute path to your app's aab/apk/ipa file
       #
       # Returns the release_id of the uploaded release.
       #
       # Crashes if the number of polling retries exceeds MAX_POLLING_RETRIES or if the binary cannot
       # be uploaded.
-      def upload(project_number, app_id, binary_path, platform)
+      def upload(app_id, binary_path, platform)
         binary_type = binary_type_from_path(binary_path)
 
-        upload_token = get_upload_token(project_number, app_id, binary_path)
-        upload_status_response = get_upload_status(app_id, upload_token)
-        if upload_status_response.success? || upload_status_response.already_uploaded?
-          UI.success("✅ This #{binary_type} has been uploaded before. Skipping upload step.")
-        else
-          unless upload_status_response.in_progress?
-            UI.message("⌛ Uploading the #{binary_type}.")
-            upload_binary(app_id, binary_path, platform)
-          end
-          MAX_POLLING_RETRIES.times do
-            upload_status_response = get_upload_status(app_id, upload_token)
-            if upload_status_response.success? || upload_status_response.already_uploaded?
-              UI.success("✅ Uploaded the #{binary_type}.")
-              break
-            elsif upload_status_response.in_progress?
-              sleep(POLLING_INTERVAL_SECONDS)
+        UI.message("⌛ Uploading the #{binary_type}.")
+        operation_name = upload_binary(app_id, binary_path, platform)
```

Comment:
The old code calls `get_upload_status` before attempting a new upload as an optimization to detect duplicate uploads.

To restore this functionality, your new code needs to call `get_operation_name` and `get_upload_status`  before calling `upload_binary`.

Reply:
I think we'd have to handle a 404 from `get_upload_status`, so it would be a little more complicated. Although I do see the appeal in saving an upload if it's not needed.

Reply:
I think we probably want to keep that functionality (even though it breaks some abstraction, because we construct the operation name from the sha-256). So you'll need to add the error handling.

Or (alternatively) we decide we don't want that optimization any more - in which case we should remove it from all three CLI clients.

Reply:
I've chatted with @lfkellogg and we agreed to remove this optimization. So you can remove `get_operation_name` since it won't be needed. I'll also remove this functionality from the Firebase CLI and we should remove it from gradle as well...

Reply:
Sounds good, done.